File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> m;
        for(int i=0; i < nums.size(); ++i) {
            m[nums[i]]++;
            if(m[nums[i]] > 1) {
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.rs
impl Solution {
    pub fn product_except_self(nums: Vec<i32>) -> Vec<i32> {
        let mut prefix: Vec<i32> = vec![1];
        let mut suffix: Vec<i32> = vec![1];
        let mut res = Vec::new();

        let mut sum: i32 = 1;
        for n in nums.iter() {
            sum *= n;
            prefix.push(sum);
        }

        sum = 1;
        for n in nums.iter().rev() {
            sum *= n;
            suffix.push(sum);
        }

        for i in 0..nums.len() {
            res.push(prefix[i] * suffix[nums.len() - i - 1]);
        }
        res
    }
}